core/store/src/deserialized_column.rs

```diff
@@ -41,7 +44,19 @@ impl ColumnCache {
     fn new(capacity: usize) -> Option<Mutex<Self>> {
         let capacity = NonZeroUsize::new(capacity);
         let values = capacity.map(|cap| lru::LruCache::new(cap))?;
-        Some(Mutex::new(Self { values, active_flushes: 0 }))
+        Some(Mutex::new(Self { values, active_flushes: 0, store_none_values: false }))
+    }
+
+    fn with_none_values(inner: Option<Mutex<Self>>) -> Option<Mutex<Self>> {
```

Comment:
there are some alternatives to this like a bool / enum param in the constructor.
btw, I'm not sure why the cache type is `Option<...>`? Since `disabled` also returns `Some(...)`

Reply:
`disabled` does not return `Some`. `NonZeroUsize::new(0)` is a `None` which we then question mark out of `new`.